packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -53,6 +54,11 @@ export class AwsCliCompatible {
       () => new AWS.EnvironmentCredentials('AMAZON'),
     ];
 
+    if (process.env.AWS_PROFILE) {
+      await forceSdkToReadConfigIfPresent();
+      sources.push(() => new AWS.SsoCredentials({ profile: implicitProfile }));
+    }
+
     if (await fs.pathExists(credentialsFileName())) {
```

Comment:
Can you try it without this `if`, and putting the `SsoCredentials` inside the `sources` array below?

Reply:
@rix0rrr yep, this change works, with both an existing (but empty) credentials file and with no credentials file. Removing the config file breaks it, but that's expected.